staging/src/k8s.io/component-base/logs/registry.go

```diff
@@ -32,21 +32,30 @@ var logRegistry = NewLogFormatRegistry()
 // LogFormatRegistry store klog format registry
 type LogFormatRegistry struct {
 	registry map[string]logr.Logger
+	freezed  bool
 }
 
 // NewLogFormatRegistry return new init LogFormatRegistry struct
 func NewLogFormatRegistry() *LogFormatRegistry {
 	return &LogFormatRegistry{
 		registry: make(map[string]logr.Logger),
+		freezed:  false,
 	}
 }
 
 // Register new log format registry to global logRegistry
 func (lfr *LogFormatRegistry) Register(name string, logger logr.Logger) error {
+	if lfr.freezed {
+		return fmt.Errorf("log format is freezed, unable to register log format")
+	}
 	if _, ok := lfr.registry[name]; ok {
 		return fmt.Errorf("log format: %s already exists", name)
 	}
 	lfr.registry[name] = logger
+
+	// Freeze log format registry
+	// to make it unable to registy new log formats
+	lfr.freezed = true
```

Comment:
I think feezing should be done after --logging-format flag calls Apply, you can add Freeze function to Register and call it in Apply https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/component-base/logs/options.go#L69